### ISSUE TITLE
operators and trends not working anymore

### ISSUE THREAD

**Issue by mimmobordini** [CNDRD/siegeapi#33]
Hi, i cannot retrieve anymore some data using load_operators() and load_trends(), bot calls seems to raise the following exception:

"ERROR - Received a text response, expected JSON response. Message: "

I tried both apis on version 6.2.5 and 6.3.4 but had the same result.

**Comment by RayanSaeed** [comment]
Same issue with `load_weapons()` now as well

**Comment by CNDRD** [comment]
Seems like Ubisoft is pulling the endpoints offline.
Well not offline, they are there but all of them are all of a sudden returning a 410 response.

Currently i don't know any more than you do. I've asked around to see if anyone knows anything but so far nothing.
All we can hope for is they will pull their heads from their asses and do some proper new endpoints since the official stats page has been _under maintenance_ for a pretty long time now..

We'll just have to wait and see

**Comment by JulianusIV** [comment]
well if you look at ubis own stat page it says its under maintenance for weeks now
https://www.ubisoft.com/en-us/game/rainbow-six/siege/stats
![image](https://github.com/user-attachments/assets/ff2c0ebe-87e6-413c-bdb0-116ecf9a6622)

**Comment by RayanSaeed** [comment]
![image](https://github.com/user-attachments/assets/d0f79542-45e6-407d-a959-d160d2e133fe)

@CNDRD Do you think they will just shut down their Stats website and its endpoints?

Or will they come out of it eventually and fix their shit?

**Comment by CNDRD** [comment]
@RayanSaeed I'm hopeful they've shut down only these specific endpoints in anticipation of the new ones. 
It's also possible they moved them to a different url.
Either way, until either someone tells me what the new endpoints are or the new stats page gets updated I'm not doing jack shit

**Comment by RayanSaeed** [comment]
That's too bad man :( 

I really hope they come back soon and fix their API asap.

**Comment by RayanSaeed** [comment]
@CNDRD Any updates on this?

Also, throughout this time, R6 TrackerNetwork continues functioning properly and providing current updated data.

**Comment by CNDRD** [comment]
@RayanSaeed nope

Didn't look into it very much tho.
TRN has people that get paid to find out. From what i know they also may have contacts directly at Ubi providing them support and shit like that (idk, just what I've heard)

Apart from the rank I've never seen very accurate data from TRN compared to Ubi's data taken directly from them there was always something off - over the years they've grown to be only really useful for the old ranks..

**Comment by RayanSaeed** [comment]
@CNDRD Did you see the latest season reveal panel? 
Ubi is bringing stats to the game itself.

I hope they will also fix the stats website and that they will start using the new endpoints on their website and finally take it out of maintenance.

**Comment by CNDRD** [comment]
@RayanSaeed I just realized I didn't actually watch the reveal panel. lol
However i am aware of the stats page in-game.
Patiently waiting for the new season to drop with some juicy new endpoints 😄

**Comment by RayanSaeed** [comment]
Yes, me too.. Waiting for the new season to drop!

**Comment by ryne007** [comment]
<img width="882" alt="image" src="https://github.com/user-attachments/assets/001f2644-4a45-4e7c-84b8-9ecb69c0ccdb">

I was just trying to run an docker image where I wrapped this package and getting the above error.. , is the API working or is it broken from ubisoft side itself?

**Comment by ARINC-717** [comment]
> I was just trying to run an docker image where I wrapped this package and getting the above error.. , is the API working or is it broken from ubisoft side itself?

Ubisoft has moved its endpoints since the last version of this module.

**Comment by CNDRD** [comment]
@RayanSaeed so yeah the new season dropped....
I kinda didn't realize that when they started encrypting the game files and shit to fend off cheaters, they would also fuck with my attempts to get anything as well. So I'm fucked on that front.
Stats section is also nowhere to be found.
And a new stats website is also still not here (probably gonna come with the stats section, if ever..)

So all in all it's fucked 🥲

**Comment by RayanSaeed** [comment]
@CNDRD Hey! 
is the status on this still #ubisoft-fucked-up? 😓

**Comment by CNDRD** [comment]
@RayanSaeed status is more like "Ubisoft removed everything useful and only shoved the info about the new APIs to the big sites"
I'm giving it a chance (a very low, low one) until the Y10 reveal.. I've heard there's gonna be quite a lot of surprisingly good stuff from a source that loves to fuck with Ubisoft big time.
If nothing changes with these APIs though, I'll finally make an update to this library to remove the not-working methods and everything else..

**Comment by RayanSaeed** [comment]
Oh wow, that's exciting news!
So this means I can pause my project for a while and come back to this later with Y10 reveal 👏🏼 

I would love to continue building and not waste all my work on the project so far. 

Let me know if you need a hand in anything @CNDRD

**Comment by CNDRD** [comment]
@RayanSaeed I wouldn't get my hopes that high up.. When I said there's exciting stuff coming in Y10 I didn't mean specifically for us developers. More like an all around game update i guess? Idk the details, dude's pretty secretive for obvious reasons, but I know him for a while and when he says it's exciting I believe him (I mean what else do we have left other than hope...).

Thanks for the offer, if anything comes up I'll be sure to reach out
o7

**Comment by RayanSaeed** [comment]
Yup, I get it but it’s tough not to get excited. Their stats site has been down for a long time.  
I hope they make an API for developers.  

Can your contact at Ubi push for a developer API? Maybe start a feature request?  

Also, I had an idea. What if I made an Overwolf app? My mobile app users (who play on PC) could download it. We could track their stats that way.  
Do you think this is a good solution? Is it even possible?

**Comment by CNDRD** [comment]
Just to be super clear, it's not a contact _at_ Ubisoft. But he is trying..

Stuff like that kinda works. Sites that used to show you who you were playing against are using it.
Like https://r6.monster/ - they have something called the 'Overwolf Listener' that works since Overwolf has access to your match data and user IDs.

**Comment by RayanSaeed** [comment]
Ahan I see.. got it 👍 

So I can make an Overwold app then.. got it

**Comment by Jasper1229** [comment]
The endpoints have to exist somewhere, otherwise how does something like r6 tracker work? Has anyone found endpoints yet?

**Comment by CNDRD** [comment]
@Jasper1229 i poke a question around every few weeks but nothing..
New official tracker is still nowhere to be seen ☹
I am unable to take the game apart to see the endpoints anymore so who knows..

My timeline has shifted so I'm giving it until Siege X happens because that thing is gonna be fucking massive..
If nothing happens then I'll make a new version of the lib that will feature only the stuff that works as well as a bunch of other much needed love that it deserves.

And yes, sites like TRN have these endpoints. AFAIK they got them from someone at Ubi.
Currently all of my contacts know jack shit about any useful endpoints, which means there isn't anything to know lol

**Comment by Jasper1229** [comment]
@CNDRD Yeah I tried contacting TRN as well, and they were pretty secretive about the endpoints. IDK if they have some sort of back channel deal with ubi or something

**Comment by nemovouches** [comment]
@CNDRD is this possible to make into a website like the old siegeskins before they got c&d? just curious if i remove all stats data.

**Comment by SwiftCODA** [comment]
Subscribing to this thread. Also been playing the waiting game with this whole thing. Super frustrating. Any new info since March?

**Comment by CNDRD** [comment]
Just today they've announced that there will be a _proper_ stats page **in-game**..
Idk what that means for the online page which is still labeled as "under construction" or whatever.
However just this past week they've absolutely nuked the siege appid and you're now unable to gather stats on even like 7 accounts at a time getting IP limited.. (You can just use proxies tho, but man fuck that...)
So apart from the in-game stats page there's really only been bad news..

**Comment by SwiftCODA** [comment]
Yea ik the siege appId got nuked for account creation. I haven't played with any of their endpoints much recently just because of the shitstorm this company is. Honestly, since they stopped updating the pre-2023 endpoints and moved everything to datadev/kong, the rate limits have been ridiculous and unpredictable. Just have to be committed enough to adapt to that unfortunately.

In-game stats sounds promising though. That means the data's coming from somewhere... only a matter of time before one of us finds where that is. It's interesting how anti-stats they are.

**Comment by SwiftCODA** [comment]
This looks promising

https://www.ubisoft.com/en-ca/game/rainbow-six/siege/news-updates/roadmap

<img width="420" alt="Image" src="https://github.com/user-attachments/assets/50cd4eca-b7ce-4afe-8918-38bf9b66683c" />

**Comment by CNDRD** [comment]
Same.. same..
Other appids are still usable for some endpoints, but the important ranked data is not accessible through them..

I mean look at TRN or stats.cc they gotta have some backdoor access or something because there's no way a site this big could adapt so fast to Ubi's bullshit 🙄

I was safe from it until now, checking literally 8 (eight) accounts, just me and my friends, and comparing our stats on a page just for shits and giggles.. But even that is now fucked..

---

Yeah the stats tracker looks good. I love that finally after 10 years they are no longer pushing KD as the main stat.

Overall I'm gonna wait until the new season and see. If nothing happens I'll give up on them completely and just cobble up something that will work for me because this has no point..

Also like if they made an official dev portal and just let us all use it with some sensible limits the world would be such a better place...

**Comment by RayanSaeed** [comment]
> Also like if they made an official dev portal and just let us all use it with some sensible limits the world would be such a better place...

Indeed! that would be ideal.

**Comment by RayanSaeed** [comment]
Now that the new Season is out with the Stats in-game. 
Do you think we now have a chance of finding the new endpoints?

**Comment by CNDRD** [comment]
@RayanSaeed 

I mean yeah 😄
If you could capture requests from the game, then yea. I don't know how to do that though 🤔

One thing I realized only yesterday is that the game shows your past three ranks which is interesting because since Ranked 2.0 we stopped having proper info about old seasons ☹
But that means they are storing that info somewhere accessible (hopefully not locked behind the special game token shit..)

**Comment by SwiftCODA** [comment]
Has anyone done some sniffing on the in-game network traffic? I've previously used Postman as a middleman but the last time I tried to do that, Siege wasn't behaving very well.

**Comment by RayanSaeed** [comment]
Have you tried Proxyman or Postman with the latest build of Siege X?

**Comment by SwiftCODA** [comment]
> Have you tried Proxyman or Postman with the latest build of Siege X?

Haven't yet. Might have time this week. Really praying they're not using sockets lol.

You guys know any other active threads on GitHub?

**Comment by CNDRD** [comment]
> Really praying they're not using sockets lol.

It's Ubi, anything is possible 😭

> You guys know any other active threads on GitHub?

I sometimes check if any new Siege API repos have popped up or new threads.. And I'm afraid that this is it

**Comment by SwiftCODA** [comment]
@RayanSaeed @CNDRD @ryne007 @mimmobordini @Jasper1229 @JulianusIV @ARINC-717 If any of you are interested, I have a small-ish Discord server for my project. I have a channel for people in the r6 dev community like us where we can hopefully share info, examples, endpoints, etc in a more concise and efficient manner. You can join [here](https://discord.gg/p6PsbXSKjT). Just let me know in there that you're a dev and I'll add you to the channel.

I have some semi-new info (from earlier this year) that is a 5000 line JSON response from Ubi which has some potential but is incredibly difficult to parse. If you guys wanna see if we can interpret it together, that would be hype.

**Comment by RayanSaeed** [comment]
I just joined the server. `ajje.` is my tag on Discord
